wordpress-mobile/WordPress-iOS
# Extract performance tests

Implements the suggestion from "RFC – Dedicated test suite for slow and performance tests".

Adds a new target to run the performance tests. I used WordPress as a target app and try to keep the target configuration as simple as possible. I ended up duplicating a bit of code, which I think is acceptable in this case.

I suggest not running the performance tests on the CI, as there is no good way to compare the performance across different executions on difference machines. The performance tests are mainly useful when working on optimizations to compare the before and the after.

To test: 

- Run `pod install`
- Verify that the new WordPressPerformanceTests target runs

Measuring the impact:

```
// Before
Test Suite 'All tests' failed at 2023-12-25 14:55:09.438.
Executed 1922 tests, with 1 test skipped and 21 failures (0 unexpected) in 77.174 (78.079) seconds

// After
Test Suite 'All tests' failed at 2023-12-25 13:28:16.761.
Executed 1918 tests, with 1 test skipped and 21 failures (0 unexpected) in 54.920 (55.530) seconds
```

That's a **22.5 sec** improvement and with the [removal](https://github.com/wordpress-mobile/WordPress-iOS/pull/22284) of the outdated version, the unit test execution time will be reduced by nearly half.


## Regression Notes
1. Potential unintended areas of impact: n/a
2. What I did to test those areas of impact (or what existing automated tests I relied on): n/a
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:


UI Changes testing checklist:
